src/plugins/data/common/search/search_source/normalize_sort_request.ts

```diff
@@ -41,9 +41,13 @@ function normalize(
   const [[sortField, sortOrder]] = Object.entries(sortable);
   const order = typeof sortOrder === 'object' ? sortOrder : { order: sortOrder };
 
-  if (indexPattern && typeof indexPattern !== 'string') {
-    const indexField = indexPattern.fields.find(({ name }) => name === sortField);
-    if (indexField && indexField.scripted && indexField.sortable) {
+  if (
+    indexPattern &&
+    typeof indexPattern !== 'string' &&
+    !indexPattern.getRuntimeField(sortField)
+  ) {
+    const indexField = indexPattern.getScriptedField(sortField);
+    if (indexField && indexField.scripted) {
```

Comment:
you're sure you don't want to check for `indexField.sortable` here? because the one test removed should work if you don't remove this. without this check, when a field that's not sortable comes along, an exception is thrown.

Reply:
There are no supported scripted field types that aren't sortable. The test used a type that isn't available.

Reply:
Got it, thx. Then I think `murmurScriptedField` should be removed

Reply:
thx!

Reply:
@mattkime it seems that when I assign 'date' as scripted field type, I can run into this situation. I can't unsort his case, since the UI is no longer available. Maybe the previous implementation is safer here?

![Bildschirmfoto 2024-05-27 um 03 24 18](https://github.com/elastic/kibana/assets/463851/ca475e9e-aa0a-4738-9df0-767dfb1badcb)

Reply:
I think this is a preexisting bug - https://github.com/elastic/kibana/issues/75711

I guess I could check the type and disallow sorting if its a date.